### DISCUSSION TITLE
Listening of the collision sensor give this Info Message :"INFO:  streaming client: failed to read header:"

### DISCUSSION THREAD

**Discussion by Arch-suzuki-MB** [carla-simulator/carla#6383] [Q&A] [answered]
Hi, I'm new in the carla simulator and I try to run this code for Reinforcement learning project that I have.
here is the code :
class CarEnv:
    SHOW_CAM = SHOW_PREVIEW
    STEER_AMT = 1.0

    im_width = 512
    im_height = 512
    actor_list = []

    front_camera = None
    collision_hist = []

    def __init__(self):
        self.client = carla.Client('localhost', 2000)
        self.client.set_timeout(10.0)

        # Once we have a client we can retrieve the world that is currently
        # running.
        self.world = self.client.get_world()

        # The world contains the list blueprints that we can use for adding new
        # actors into the simulation.
        blueprint_library = self.world.get_blueprint_library()

        # Now let's filter all the blueprints of type 'vehicle' and choose one
        # at random.
        #print(blueprint_library.filter('vehicle'))
        self.model_3 = blueprint_library.filter('model3')[0]
    
    
    
    def process_img(self, image):
        i = np.array(image.raw_data)
        #np.save("iout.npy", i)
        i2 = i.reshape((self.im_height, self.im_width, 4))
        i3 = i2[:, :, :3]
        if self.SHOW_CAM:
            cv2.imshow("",i3)
            cv2.waitKey(1)
        self.front_camera = i3
   
   
   
    def reset(self):
        self.collision_hist = []
        self.actor_list = []

        self.transform = random.choice(self.world.get_map().get_spawn_points())
        self.vehicle = self.world.spawn_actor(self.model_3, self.transform)
        self.actor_list.append(self.vehicle)

        self.rgb_cam = self.world.get_blueprint_library().find('sensor.camera.rgb')

        self.rgb_cam.set_attribute('image_size_x', f'{self.im_width}')
        self.rgb_cam.set_attribute('image_size_y', f'{self.im_height}')
        self.rgb_cam.set_attribute('fov', '110')

        transform = carla.Transform(carla.Location(x=2.5, z=0.7))

        self.sensor = self.world.spawn_actor(self.rgb_cam, transform, attach_to=self.vehicle)

        self.actor_list.append(self.sensor)
        self.sensor.listen(lambda data: self.process_img(data))

        self.vehicle.apply_control(carla.VehicleControl(throttle=0.0, brake=0.0))

        time.sleep(4) # sleep to get things started and to not detect a collision when the car spawns/falls from sky.

        colsensor = self.world.get_blueprint_library().find('sensor.other.collision')
        self.colsensor = self.world.spawn_actor(colsensor, transform, attach_to=self.vehicle)
        self.actor_list.append(self.colsensor)
        self.colsensor.listen(lambda event: self.collision_data(event))

        while self.front_camera is None:
            time.sleep(0.01)

        self.episode_start = time.time()

        self.vehicle.apply_control(carla.VehicleControl(brake=0.0, throttle=0.0))

        return self.actor_list

if __name__=="__main__":
    

    scenario=CarEnv()
    
    while True:
        actor_list=scenario.reset()
        print("Destroying the actors`\n")
        scenario.client.apply_batch([carla.command.DestroyActor(actor) for actor in actor_list])
        print("All Cleaned up!")
        

and when I run the code the simulation is fine but there is a message dispayed :

![image](https://user-images.githubusercontent.com/99597954/229886234-da0a8c97-c53f-4747-ad42-8f350b7aaf53.png)
I don't know if this message will affect the DQN agent when it will be training or it's not a critical message 
I tried other sensors like invasion sensors but It's not showing this message 
I hope that anyone  answer my question as soon as possible in order to go further in my project. and Thanks a lot for any help

**Comment by Arch-suzuki-MB** [comment]
@jackbart94 Please I need your help here

**Reply by jackbart94** [reply]
Sorry I can't be of any help, I don't work for CARLA anymore so I'm a bit behind on any SW development. I guess you should try either the Discord server or pinging any current CARLA developer. But even before that, I would suggest including the version of CARLA you're running this on and reformatting the code to make it readable. Cheers!

**Comment by Arch-suzuki-MB** [comment, marked as answer]
Thanks, I resolve my issue It was just the version 0.9.12 that I used when I upgrade the version to 0.9.13 It work fine
